app/src/main/java/com/nearit/ui_bindings_sample/CouponPlainActivity.java

```diff
@@ -0,0 +1,44 @@
+package com.nearit.ui_bindings_sample;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v7.app.AppCompatActivity;
+
+import com.nearit.ui_bindings.NearITUIBindings;
+
+import java.util.ArrayList;
+
+import it.near.sdk.reactions.couponplugin.model.Claim;
+import it.near.sdk.reactions.couponplugin.model.Coupon;
+
+/**
+ * Created by Federico Boschini on 13/09/17.
+ */
+
+public class CouponPlainActivity extends AppCompatActivity {
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_plain_coupon);
+
+        final Coupon validCoupon = new Coupon();
```

Comment:
It's better to move coupon creation and init in a method and return it.

EDIT: Just realized you are repeating the process in various activities, it's better to move the creation method in a different object, like a CouponFactory and put a comment before calling it's method saying that in a real integration scenario, the coupon object comes from the NearIT library

Reply:
done